doc/denite.txt

```diff
@@ -949,15 +951,15 @@ command		Gather commands and put on ex-mode.
 
 						*denite-source-grep*
 grep		Gather grep results and nominates them.
-		The source executes ["command", "default_opts",
-		"recursive_opts", {arguments}, "separator", "final_opts"]
+		The source executes [command, default_opts, recursive_opts, args[0]...,
+		`pattern_opt, args[1][0]`..., separator, args[2]..., final_opts]
 		command line.
 		Note: The command result must be "path:linenr:text" pattern.
-		Note: If the input contains the spaces, you must quote it.
 
-		Source arguments:
-		1. the search directory.
-		2. the additional arguments.
+		Source args (all optional):
+			0. list of additional arguments
+			1. list of patterns
+			2. list of paths
```

Comment:
Because, it cannot be set by the command line.
And you don't check the types in the souce.
Have you tested?